packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+        setProcessSuccessText(getAddSuccessMessage(language, validFiles.length, files.length));
+      }
+    }
+  };
+
+  const onFilesChange = (files: File[]) => {
+    clearNotificationTexts();
+
+    if (multiple) {
+      handleMultipleChange(files);
+    } else {
+      handleSingleFileChange(files);
+    }
+  };
+
+  const onRemoveFileFromList = (fileToRemove: File, index: number) => {
+    clearNotificationTexts();
+
+    const selectedFilesWithoutRemoved = selectedFiles.filter(
+      (file: File) => !isEqualFileBy(['name', 'type', 'size', 'lastModified'], file, fileToRemove),
+    );
+    setSelectedFiles(selectedFilesWithoutRemoved);
+    setInputStateText(getRemoveSuccessMessage(language));
+
+    if (selectedFilesWithoutRemoved.length > 0) {
+      fileListFocusIndexRef.current = index > 0 ? index - 1 : 0;
+      setInputStateText(getRemoveSuccessMessage(language));
+    } else if (inputRef.current) {
+      inputRef.current.focus();
+      setInputStateText(getNoFilesAddedMessage(language));
+    }
+  };
+
+  const onDragEnter = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(true);
+  };
+
+  const onDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(false);
+  };
+
+  useEffect(() => {
+    if (didMountRef.current && onChange) {
+      onChange(selectedFiles);
+    } else {
+      didMountRef.current = true;
+    }
+    // Clear input value on every change to ensure it triggers a onChange event when files are added
+    resetFileInputValue();
+  }, [selectedFiles, onChange]);
+
+  // Compose aria-describedby attribute
+  const ariaDescribedBy: string = [
+    composeAriaDescribedBy(id, helperTextToUse, errorTextToUse, successText, infoTextToUse),
+    hasFilesSelected && fileListId,
+  ]
+    .filter((text) => !!text)
+    .join(' ');
+
+  return (
+    <>
+      <InputWrapper {...wrapperProps}>
+        <div className={styles.fileInputContainer}>
+          {hasDragAndDrop && (
+            <>
+              <div
+                aria-hidden
+                onDragEnter={onDragEnter}
+                onDragOver={onDragEnter}
+                onDragLeave={onDragLeave}
+                onDrop={(event: React.DragEvent<HTMLDivElement>) => {
+                  const { dataTransfer }: { dataTransfer: DataTransfer } = event;
+                  onDragLeave(event);
+                  onFilesChange(Array.from(dataTransfer.files));
+                }}
+                className={classNames(styles.dragAndDrop, isDragOverDrop && styles.dragAndDropActive)}
+                ref={dropAreaRef}
+              >
+                <div className={styles.dragAndDropLabel}>
+                  <IconUpload aria-hidden />
+                  <span className={styles.dragAndDropLabelText}>{dragAndDrop.label}</span>
```

Comment:
Noticed that if you click this drag and drop label, then it will open the choose file dialog. I wonder if that is what we want? It could be nice, but then we would probably want it to be open also if user click the drag and drop area?

Reply:
- [ ] add drag-and-drop area click handler